### DISCUSSION TITLE
React Native and NetInfo's Online status management

### DISCUSSION THREAD

**Discussion by ftzi** [TanStack/query#3590] [Q&A]
In this link, we see how to setup React Native, React Query and NetInfo: https://react-query.tanstack.com/react-native#online-status-management

But, as TS typing and the https://github.com/react-native-netinfo/react-native-netinfo#types says, `state.isConnected` is `boolean | null`.

If we try to `setOnline(state.isConnected)`, it throws `Argument of type 'boolean | null' is not assignable to parameter of type 'boolean | undefined'.
  Type 'null' is not assignable to type 'boolean | undefined'`.

The link above about `isConnected`: `If there is an active network connection. Defaults to null on most platforms for unknown networks. Note: Web browsers report network type unknown for many otherwise valid networks (https://caniuse.com/netinfo), so isConnected may be true or false and represent a real connection status even for unknown network types in certain cases.`
  
What to do when the value is `null`?

**Comment by TkDodo** [comment]
> Defaults to null on most platforms for unknown networks

I have no idea what an `unknown network` would be. 🤷 

You can coerce it to `true` if you want null to be treated as online, or `false` if you want null to be treated as offline. You can also coerce to `undefined` to fall back to the react-query default, which checks for `navigator.onLine`, but that's likely not what you want for react-native...

```
setOnline(state.isConnected ?? true)
setOnline(state.isConnected ?? false)
setOnline(state.isConnected ?? undefined)
```
